### PR TITLE
Feature/group spaces in space list by space type 1973

### PR DESCRIPTION
**Pull request by Daboo1998**

### CODE REVIEW

**Review by PAKovalev — APPROVED** [review]

File: src/scenes/default/pages/HomePage/components/ExplorePanel/components/SpacesList/SpacesList.styled.ts [hunk 1]
```diff
@@ -1,6 +1,31 @@
 import styled from 'styled-components';
+import {rgba} from 'polished';
+
+import {Heading} from 'ui-kit';
 
 export const Container = styled.div`
   height: 100%;
   overflow-y: scroll;
+  ::-webkit-scrollbar {
```
Comment (PAKovalev) [inline src/scenes/default/pages/HomePage/components/ExplorePanel/components/SpacesList/SpacesList.styled.ts:9]: Can we somehow make a separate reusable component / class for that?

File: src/scenes/default/pages/HomePage/components/ExplorePanel/components/SpacesList/SpacesList.styled.ts [hunk 2]
```diff
@@ -1,6 +1,31 @@
...
+    border-radius: 6px;
+    border: 5px solid ${(props) => props.theme.accent && rgba(props.theme.accent, 0.6)};
+  }
+  ::-webkit-scrollbar-track {
+    display: none;
+  }
+  ::-webkit-scrollbar-track-piece:end {
+    margin-bottom: 10px;
+  }
+  ::-webkit-scrollbar-track-piece:start {
```
Comment (PAKovalev) [inline src/scenes/default/pages/HomePage/components/ExplorePanel/components/SpacesList/SpacesList.styled.ts:23]: empty

File: src/scenes/default/pages/HomePage/components/ExplorePanel/components/SpacesList/SpacesList.tsx [hunk 3]
```diff
@@ -11,33 +11,44 @@ const SpacesList: React.FC = () => {
...
+          <styled.CategoryName label={category.name} type="h4" align="left" />
+          {category.spaces.map((space, index) => (
+            <SpaceItem
+              space={{
+                id: space.id ?? '',
+                name: space.name ?? '',
+                hasSubspaces: space.subSpaces.length > 0
+              }}
+              hasSubspaces={space.subSpaces.length > 0}
+              key={`space-${space.id}`}
```
Comment (PAKovalev) [inline src/scenes/default/pages/HomePage/components/ExplorePanel/components/SpacesList/SpacesList.tsx:33]: just `key={space.id}`

File: src/scenes/default/pages/HomePage/components/ExplorePanel/components/SpacesList/SpacesList.tsx [hunk 4]
```diff
@@ -11,33 +11,44 @@ const SpacesList: React.FC = () => {
...
       return null;
     }
 
-    return selectedSpace.subSpaces.map((space) => (
-      <SpaceItem space={space} hasSubspaces={space.hasSubspaces} key={`space-${space.id}`} />
+    return selectedSpace.subSpaces.map((space, index) => (
+      <SpaceItem
+        space={space}
+        hasSubspaces={space.hasSubspaces}
+        key={`space-${space.id}`}
```
Comment (PAKovalev) [inline src/scenes/default/pages/HomePage/components/ExplorePanel/components/SpacesList/SpacesList.tsx:49]: just `key={space.id}`

File: src/scenes/widgets/stores/ExploreStore/models/ExploreCategoryModel/ExploreCategoryModel.ts [hunk 5]
```diff
@@ -0,0 +1,12 @@
+import {Instance, types} from 'mobx-state-tree';
+
+import {SpaceModel} from 'core/models';
+
+const ExploreCategoryModel = types.model('ExploreCategoryModel', {
+  name: types.string,
+  spaces: types.optional(types.array(types.optional(SpaceModel, {})), [])
```
Comment (PAKovalev) [inline src/scenes/widgets/stores/ExploreStore/models/ExploreCategoryModel/ExploreCategoryModel.ts:7]: spaces: types.optional(types.array(SpaceModel), [])

File: src/scenes/widgets/stores/ExploreStore/models/ExploreCategoryModel/ExploreCategoryModel.ts [hunk 6]
```diff
@@ -0,0 +1,12 @@
...
+});
+
+export interface ExploreCategoryModelInterface extends Instance<typeof ExploreCategoryModel> {}
```
Comment (PAKovalev) [inline src/scenes/widgets/stores/ExploreStore/models/ExploreCategoryModel/ExploreCategoryModel.ts:10]: Do we use this interface? If not, remove pls.

**Review by PAKovalev — CHANGES_REQUESTED** [review]

File: src/scenes/widgets/stores/ExploreStore/models/ExploreCategoryModel/ExploreCategoryModel.ts (hunk 6 above)
Comment (aminbksi) [reply src/scenes/widgets/stores/ExploreStore/models/ExploreCategoryModel/ExploreCategoryModel.ts:10]: yes for retrieving spaces based on the categories

**Pull request merged**